packages/devextreme/js/__internal/ui/html_editor/modules/m_mentions.ts

```diff
@@ -238,11 +238,21 @@ if (Quill) {
       };
       const Delta = Quill.import('delta');
       const startIndex = Math.max(0, caretPosition - markerLength);
+
+      const currentContents = this.quill.getContents();
+
+      let retainedAttributes = {};
+      currentContents.ops.forEach((op, index) => {
+        if (index <= startIndex && op.attributes) {
+          retainedAttributes = { ...retainedAttributes, ...op.attributes };
+        }
+      });
```

Comment:
[broken behavior, incorrect solution]

When u call `this.quill.getContents()` you get the whole editor content in `Delta` format including tables images and so on. Then u collect all attributes and u get smth like this 
<img width="312" alt="image" src="https://github.com/DevExpress/DevExtreme/assets/37047972/143caacc-0b14-486a-8d8e-0ed74fb93c2e">
because u got all attributes for all elements in the content.
And then you try to add all this attributes to the mention and inserted string: some of them are declined because it's not applicable, other - applied (text formats).

You can see unexpected behavior on this basic scenario:
![HtmlEditor_mention_insert](https://github.com/DevExpress/DevExtreme/assets/37047972/cd1ad0c5-23cd-40fe-b220-9659fd22c329)
Here only "bold" format should be applied, but we also see cross out at least.

I think you can just use `this.quill.getFormat(startIndex)` to get necessary formats.
And I also think we should apply this format only for newly inserted `" "` string because seems it's already applied for a mention

P.s test should be rewritten

Reply:
[349e3be](https://github.com/DevExpress/DevExtreme/pull/27636/commits/349e3be9ea27f24cdc65f16749c335876438b0bd)